driver/heap.go

```diff
@@ -0,0 +1,33 @@
+// This file contains code from these Go 1.14.5 source files:
+// - container/heap/example_intheap_test.go
+//
+// That code is covered by the following copyright notice,
+// and the license file mentioned is available at:
+// https://github.com/golang/go/blob/go1.14.5/LICENSE
+//
+// Copyright 2012 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package driver
+
+// An IntHeap is a min-heap of ints.
+type IntHeap []int
```

Comment:
Consider making this private? I'm not sure this makes sense as a public type in package driver.